haskell/defs.bzl

```diff
@@ -343,7 +342,6 @@ haskell_toolchain_library = rule(
             default = Label("@rules_haskell//haskell:toolchain-libraries"),
         ),
     ),
-    fragments = ["cpp"],
```

Comment:
Why `fragment` was removed in this rule but not the others?

Reply:
I haven't removed `fragment` and `current_cc_toolchain`, yet, so that rules_haskell still works without setting `incompatible_enable_cc_toolchain_resolution`.

This instance was removed because it is not needed. The implementation of `haskell_toolchain_library` only forwards some providers and doesn't lookup the cc toolchain.